projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -345,5 +312,20 @@ private void ConnectOrFail(ITcpClient socket, IPEndPoint endpoint, TimeSpan time
                 throw new ConnectFailureException("Connection failed", e);
             }
         }
+
+        public Span<byte> GetSpan(int sizeHint)
+        {
+            return _pipeWriter.GetSpan(sizeHint);
+        }
+
+        public void Commit(int bytesToCommit)
+        {
+            _pipeWriter.Advance(bytesToCommit);
+            ValueTask<FlushResult> task = _pipeWriter.FlushAsync();
+            if (!task.IsCompleted)
+            {
+                task.AsTask().Wait();
```

Comment:
use at least getawaiter().getResult() to unwrap the correct exception in case of an error

Reply:
You are right, I'll fix that :)